### ISSUE TITLE
panic in CommitConfiguration

### ISSUE THREAD

**Issue by canton7** [syncthing/syncthing#2788]
Version 0.12.17. Tried to save the configuration using Syncthing's web GUI (disabled HTTPS, and added a username and password), and was treated to this panic in the console.

Everything still appeared to function (although the configuration was not saved).

Can't reproduce.

```
2016/02/18 20:18:50 http: panic serving 127.0.0.1:55911: runtime error: invalid memory address or nil pointer dereference
goroutine 275 [running]:
net/http.(*conn).serve.func1(0xc082680280)
	c:/go/src/net/http/server.go:1389 +0xc8
main.(*apiService).CommitConfiguration(0xc08278cb60, 0xc, 0xc082ad2c00, 0x6, 0x6, 0xc0824fe680, 0x4, 0x4, 0x1, 0xc08377ff30, ...)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:325 +0x145
github.com/syncthing/syncthing/lib/config.(*Wrapper).replaceLocked(0xc082181180, 0xc, 0xc082110000, 0x6, 0x6, 0xc08256e1a0, 0x4, 0x4, 0x1, 0xc08377faa0, ...)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/lib/config/wrapper.go:157 +0x73b
github.com/syncthing/syncthing/lib/config.(*Wrapper).Replace(0xc082181180, 0xc, 0xc082110000, 0x6, 0x6, 0xc08256e1a0, 0x4, 0x4, 0x1, 0xc08377faa0, ...)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/lib/config/wrapper.go:138 +0xfa
main.(*apiService).postSystemConfig(0xc08278cb60, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:671 +0x490
main.(*apiService).(main.postSystemConfig)-fm(0x3363c18, 0xc087f4bc70, 0xc084612380)
[UAL5T] 20:18:50 INFO: Completed initial scan (rw) of folder Docs
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:200 +0x45
net/http.HandlerFunc.ServeHTTP(0xc0834061a0, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1618 +0x41
net/http.(*ServeMux).ServeHTTP(0xc0823a8930, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1910 +0x184
main.getPostHandler.func1(0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:348 +0x146
net/http.HandlerFunc.ServeHTTP(0xc0823a8960, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1618 +0x41
main.metricsMiddleware.func1(0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:416 +0xd8
net/http.HandlerFunc.ServeHTTP(0xc0837ba300, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1618 +0x41
main.noCacheMiddleware.func1(0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:439 +0x1e3
net/http.HandlerFunc.ServeHTTP(0xc0837ba320, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1618 +0x41
net/http.(*ServeMux).ServeHTTP(0xc0823a89c0, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1910 +0x184
main.csrfMiddleware.func1(0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui_csrf.go:38 +0x169
net/http.HandlerFunc.ServeHTTP(0xc0823b1db0, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1618 +0x41
main.corsMiddleware.func1(0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:407 +0x212
net/http.HandlerFunc.ServeHTTP(0xc0822c90a0, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1618 +0x41
main.withDetailsMiddleware.func1(0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:447 +0x12e
net/http.HandlerFunc.ServeHTTP(0xc08265fb00, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1618 +0x41
main.debugMiddleware.func1(0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/jenkins/workspace/syncthing-release-windows/src/github.com/syncthing/syncthing/cmd/syncthing/gui.go:358 +0xa2
net/http.HandlerFunc.ServeHTTP(0xc0822c9100, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:1618 +0x41
net/http.serverHandler.ServeHTTP(0xc08222a600, 0x3363c18, 0xc087f4bc70, 0xc084612380)
	c:/go/src/net/http/server.go:2081 +0x1a5
net/http.(*conn).serve(0xc082680280)
	c:/go/src/net/http/server.go:1472 +0xf35
created by net/http.(*Server).Serve
	c:/go/src/net/http/server.go:2137 +0x455
```

**Comment by lkwg82** [comment]
I'll close this, as nobody else had this issue and even the reporter does not know how to reproduce.

**Comment by canton7** [comment]
It's a panic. Whether or not anyone else had commented on this issue, a nil pointer dereference is a definite, real, tangible issue. I'm not sure that blithely closing it off because noone's looked at it is the right thing to do...

**Comment by AudriusButkevicius** [comment]
Yeah, agree

**Comment by canton7** [comment]
Cool! Yeah it's not fatal, and definitely low priority, but there's a difference between a something that isn't a bug and a low-priority bug.

**Comment by AudriusButkevicius** [comment]
I suspect this is fixed by 5aacfd16395efc6a375e21113ded6cb3b0d5783e
